### PR TITLE
pyproject.toml: New

### LINKED ISSUE #626: Declare build dependencies in pyproject.toml
Building a source distribution using `python3 -m build --sdist .` fails because PySCIPOpt does not declare its PEP-518 build dependencies using `pyproject.toml`.
```
* Creating venv isolated environment...
* Installing packages in isolated environment... (setuptools >= 40.8.0, wheel)
* Getting build dependencies for sdist...
Assuming that SCIP is installed globally, because SCIPOPTDIR is undefined.
Cython is required.
```
https://github.com/mkoeppe/PySCIPOpt/actions/runs/3452468222/jobs/5762391556

### PR DESCRIPTION
**Pull request by mkoeppe**
Fixes #626

### CODE REVIEW

**Comment by mkoeppe** [comment]
@mattmilten @wkschwartz ping

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[build-system]
+requires = ['setuptools', 'cython <3']
```
Comment (wkschwartz) [inline pyproject.toml:2]: Instructions you (correctly) deleted above said `cython >= .21`. If the `< 3` constraint is also correct (I don't know what pyscipopt needs so I have no opinion other than that the old instructions don't mention it) then you probably want
```suggestion
requires = ['setuptools', 'cython <3, >=0.21']
```

**Review by wkschwartz — COMMENTED** [review]
I don't have enough familiarity with pyscipopt to opine on the list of requirements itself beyond what I see in this patch set

File: pyproject.toml (hunk 1 above)
Comment (mkoeppe) [reply pyproject.toml]: Yes, adding this is also fine and correct; but setting this lower bound is not really needed – these build system requirements are resolved in their own venv, and there won't be a conflict with other packages that would force a lower cython version into this venv.

Cython < 3 protects against the untested major Cython release that has been in the making for many years.

**Comment by mattmilten** [comment]
Thanks a lot!

**Pull request merged**